Xamarin.Forms.Core/Page.cs

```diff
@@ -379,7 +379,7 @@ protected void UpdateChildrenLayout()
 				var element = LogicalChildren[i];
 				if (element is VisualElement c)
 				{
-					if (c.Bounds != startingLayout[i])
+					if (startingLayout.Count <= i || c.Bounds != startingLayout[i])
```

Comment:
Now that generated flyout items are added to the LogicalChildren the `startingLayout.Count` no longer matches up with `LogicalChildren.Count` after LayoutChildren is called